fdbserver/workloads/BackupRestoreDemo.actor.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * BackupRestoreDemo.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cstdint>
+#include <limits>
+#include "fdbrpc/simulator.h"
+#include "fdbclient/BackupAgent.actor.h"
+#include "fdbclient/BackupContainer.h"
+#include "fdbclient/MutationLogReader.actor.h"
+#include "fdbserver/workloads/workloads.actor.h"
+#include "flow/Error.h"
+#include "flow/IRandom.h"
+#include "flow/flow.h"
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+struct BackupRestoreDemoWorkload : TestWorkload {
+	bool enabled;
+
+	BackupRestoreDemoWorkload(WorkloadContext const& wcx) : TestWorkload(wcx) {
+		enabled = !clientId; // only do this on the "first" client
+	}
+
+	std::string description() const override { return "BackupRestoreDemo"; }
+
+	Future<Void> start(Database const& cx) override {
+		if (enabled) {
+			return _start(cx, this);
+		}
+		return Void();
+	}
+
+	ACTOR Future<Void> _start(Database cx, BackupRestoreDemoWorkload* self) {
+		state Transaction tr(cx);
+
+		try {
+			std::cout << "start tr" << std::endl;
+			tr.reset();
+			tr.clear(KeyRangeRef("hello"_sr, "world"_sr), AddConflictRange::True, MutationRef::Type::RestoreRange);
```

Comment:
@sears and Chris, Is this the API you want?
Should the restore intent tell which range chucker or url to fetch the data?

Reply:
I checked the code again. It turns out hacking the version into the key might be hard, because `clearRange` path uses the key to calculate the locations. Looks like adding `param3` at https://github.com/apple/foundationdb/pull/5573/files#diff-bb3f807cef7c8b7ade7626aa75c03813451c9d041bfe99a59b75018476ef034cR88 might be easiest way (and probably the correct way.)

Reply:
We'd have to be fantastically unlucky for a boundary to fall between the target key and the key+version, but if we can add another parameter then I agree: that's better w/ no correctness edge cases.